src/libs/xen-api.ts

```diff
@@ -190,6 +194,41 @@ export default class XenApi {
     return this.#client.request(method, args);
   }
 
+  async getHostServertime(host: XenApiHost) {
+    const serverLocaltime = (await this.#call("host.get_servertime", [
+      this.sessionId,
+      host.$ref,
+    ])) as string;
+    return Math.floor(this.parseDateTime(serverLocaltime) / 1e3);
+  }
+
+  async getResource(
+    pathname: string,
+    { host, query }: { host: XenApiHost; query: any }
+  ) {
+    const url = new URL("http://localhost");
+    url.protocol = window.location.protocol;
+    url.hostname = host.address;
+    url.pathname = pathname;
+    url.search = new URLSearchParams({
+      ...query,
+      session_id: this.#sessionId,
+    }).toString();
+
+    const response = await fetch(url);
+    return JSON5.parse(await response.text());
```

Comment:
getResource is used  to get non JSON content ( like a full VHD ) . I think you should handle it , or at least add a comment